sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -462,6 +462,34 @@ def is_warm_reboot_enabled():
     is_warm_start = warmstart.isWarmStart()
     return is_warm_start
 
+def is_syncd_warm_restore_complete(self):
+    """
+    This function determins whether syncd's restore count is not 0, which indicates warm-reboot
+    to avoid premature config push by xcvrd that caused port flaps.
+    """
+    state_db = daemon_base.db_connect("STATE_DB")
+    restore_count = state_db.hget("WARM_RESTART_TABLE|syncd", "restore_count")
+    system_enabled = state_db.hget("WARM_RESTART_ENABLE_TABLE|system", "enable")
+    try:
+        # --- Handle restore_count (could be int, str, or None) ---
+        if restore_count is not None:
+            if isinstance(restore_count, int):
+                if restore_count > 0:
+                    return True
+            elif isinstance(restore_count, str):
+                if restore_count.strip().isdigit() and int(restore_count.strip()) > 0:
+                    return True
+
+        # --- Handle system_enabled (only care about "true"/"false"/None) ---
+        if isinstance(system_enabled, str):
+            if system_enabled.strip().lower() == "true":
+                return True
+
+    except Exception as e:
+        self.log_warning(f"[WARN] Unexpected value: restore_count={restore_count}, system_enabled={system_enabled}, error={e}")
```

Comment:
This still does not look right or does not represent the change in master.

The changes on both branches should be same.